tests/Fixer/Whitespace/BlankLineBeforeControlStatementFixerTest.php

```diff
@@ -0,0 +1,1085 @@
+while (true) {
+    if ($foo === $bar) {
+        echo $baz;
+        break;
+    }
+}',
+            ],
+            [
+'<?php
+while (true) {
+    if ($foo === $bar) {
+        echo $baz;
+
+        break 1;
+    }
+}',
+'<?php
+while (true) {
+    if ($foo === $bar) {
+        echo $baz;
+        break 1;
+    }
+}',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerFixWithContinue
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithContinue($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'statements' => ['continue'],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithContinue()
+    {
+        return [
+            [
+'<?php
+while (true) {
+    continue;
+}',
+            ],
+            [
+'<?php
+while (true) {
+    continue 1;
+}',
+            ],
+            [
+'<?php
+while (true) {
+    while (true) {
+        continue 2;
+    }
+}',
+            ],
+            [
+'<?php
+while (true) {
+    $foo = true;
+
+    continue;
+}',
+'<?php
+while (true) {
+    $foo = true;
+    continue;
+}',
+            ],
+            [
+'<?php
+while (true) {
+    $foo = true;
+
+    continue 1;
+}',
+'<?php
+while (true) {
+    $foo = true;
+    continue 1;
+}',
+            ],
+            [
+'<?php
+while (true) {
+    while (true) {
+        $foo = true;
+
+        continue 2;
+    }
+}',
+'<?php
+while (true) {
+    while (true) {
+        $foo = true;
+        continue 2;
+    }
+}',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerFixWithDeclare
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithDeclare($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'statements' => ['declare'],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithDeclare()
+    {
+        return [
+            [
+'<?php
+declare(ticks=1);',
+            ],
+            [
+'<?php
+
+$foo = "bar";
+
+declare(ticks=1);',
+'<?php
+
+$foo = "bar";
+declare(ticks=1);',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerFixWithDo
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithDo($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'statements' => ['do'],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithDo()
+    {
+        return [
+            [
+'<?php
+do {
+} while (true);',
+            ],
+            [
+'<?php
+$foo = "bar";
+
+do {
+} while (true);',
+'<?php
+$foo = "bar";
+do {
+} while (true);',
+            ],
+        ];
```

Comment:
I think this one still needs to be addressed. Some tests don't cover cases like:
```php
if ($foo) {
    for ($i = 0; $i < 9000; ++$i) {
        foo();
    }
}
```
which should not be changed, i.e. assert that the fixer does not add a newline before the `for` loop because it is at the beginning of a control structure.